common/app/common/metrics.scala

```diff
@@ -39,384 +32,151 @@ object SystemMetrics extends implicits.Numbers {
   )
 
-  object FreeDiskSpaceMetric extends GaugeMetric("free-disk-space", "Free disk space (MB)",
-    () => new File("/").getUsableSpace / 1048576
-  )
-
-  object TotalDiskSpaceMetric extends GaugeMetric("total-disk-space", "Total disk space (MB)",
-    () => new File("/").getTotalSpace / 1048576
-  )
-
-  object ThreadCountMetric extends GaugeMetric("thread-count", "Thread Count",
-    () => ManagementFactory.getThreadMXBean.getThreadCount,
-    StandardUnit.Count
+  val ThreadCountMetric = GaugeMetric(
+    name = "thread-count",
+    description = "Thread Count",
+    get = () => ManagementFactory.getThreadMXBean.getThreadCount,
+    metricUnit = StandardUnit.Count
   )
 
   // yeah, casting to com.sun.. ain't too pretty
-  object TotalPhysicalMemoryMetric extends GaugeMetric("total-physical-memory", "Total physical memory",
-    () => ManagementFactory.getOperatingSystemMXBean match {
+  val TotalPhysicalMemoryMetric = GaugeMetric(
+    name = "total-physical-memory", description = "Total physical memory",
+    get = () => ManagementFactory.getOperatingSystemMXBean match {
       case b: com.sun.management.OperatingSystemMXBean => b.getTotalPhysicalMemorySize
       case _ => -1
     }
   )
 
-  object FreePhysicalMemoryMetric extends GaugeMetric("free-physical-memory", "Free physical memory",
-    () => ManagementFactory.getOperatingSystemMXBean match {
+  val FreePhysicalMemoryMetric = GaugeMetric(
+    name = "free-physical-memory", description = "Free physical memory",
+    get = () => ManagementFactory.getOperatingSystemMXBean match {
       case b: com.sun.management.OperatingSystemMXBean => b.getFreePhysicalMemorySize
       case _ => -1
     }
   )
 
-  object OpenFileDescriptorsMetric extends GaugeMetric("open-file-descriptors", "Open file descriptors",
-    () => ManagementFactory.getOperatingSystemMXBean match {
-      case b: com.sun.management.UnixOperatingSystemMXBean => b.getOpenFileDescriptorCount
-      case _ => -1
-    }
-  )
-
-  object MaxFileDescriptorsMetric extends GaugeMetric("max-file-descriptors", "Max file descriptors",
-    () => ManagementFactory.getOperatingSystemMXBean match {
-      case b: com.sun.management.UnixOperatingSystemMXBean => b.getMaxFileDescriptorCount
-      case _ => -1
-    }
-  )
-
   private lazy val buildNumber = ManifestData.build match {
     case string if string.isInt => string.toInt
     case _ => -1 // dev machines do not have a build number
   }
 
-  object BuildNumberMetric extends GaugeMetric("build-number", "Build number",
-    () => buildNumber,
-    StandardUnit.None
-  )
-}
-
-object S3Metrics {
-  object S3ClientExceptionsMetric extends CountMetric(
-    "s3-client-exceptions",
-    "Number of times the AWS S3 client has thrown an Exception"
-  )
-
-  object S3AuthorizationError extends CountMetric(
-    "s3-authorization-errors",
-    "Number of requests to S3 by facia that have resulted in a 403"
+  val BuildNumberMetric = GaugeMetric(
```

Comment:
I've used it to work out when a build was live, using average. Bit like riffraff, except you can overlay the graph on another metric, and correlate step changes with deployments. Build number usually looks like an ascending staircase (the exception being reverts)